Source/Core/Common/CommonFuncs.cpp

```diff
@@ -2,32 +2,59 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
-#include <cstddef>
+// The code in GetErrorMessage can't handle some systems having the
+// GNU version of strerror_r and other systems having the XSI version,
+// so we undefine _GNU_SOURCE here in an attempt to always get the XSI version.
+// We include cstring before all other headers in case cstring is included
+// indirectly (without undefining _GNU_SOURCE) by some other header.
+#ifdef _GNU_SOURCE
+#undef _GNU_SOURCE
 #include <cstring>
-#include <errno.h>
+#define _GNU_SOURCE
+#else
+#include <cstring>
+#endif
+
+#include <cstddef>
 
 #include "Common/CommonFuncs.h"
 
 #ifdef _WIN32
 #include <windows.h>
+#else
+#include <errno.h>
 #endif
 
-// Generic function to get last error message.
-// Call directly after the command or use the error num.
-// This function might change the error code.
-std::string GetLastErrorMsg()
+int GetLastErrorCode()
+{
+#ifdef _WIN32
+  return GetLastError();
```

Comment:
This is kind of misleading: Windows still has `errno`, and it's used for the same CRT/POSIX-y things as on non-windows. `GetLastError` is for win32 apis.